### DISCUSSION TITLE
Maximum number of characters/words per caption as input argument

### DISCUSSION THREAD

**Discussion by Francoyy** [openai/whisper#1482] [Ideas]
There is no control on how long a subtitle can be.
Unfortunately it seems that it's very random. It would be good to have a maximum number of characters as input argument:

`--max_characters 62`

This would give much more control on the expected size of the output.

For example, at the beginning of my video I have the following:

```
[04:10.540 --> 04:15.740]  Anyway, I did want to point out a few things before you actually bring your water over to Taipei Water Department.
[04:16.220 --> 04:20.200]  And that is, of course, that you need to keep the water running for three minutes before you take the sample.
```

But most of the other subtitles are much shorter, like this:

```
[08:29.460 --> 08:31.040]  This is a sparkling water
[08:31.040 --> 08:33.360]  machine, so it will actually add sparkles
[08:33.360 --> 08:35.300]  or carbon dioxide to your water
```

The first ones are way too long, and will require additional work to be burned-in the video. I would need to manually cut the long sentences in half.
Hence this request for a new argument to control the maximum length.

**Comment by ryanheise** [comment]
Try:

`--output_format srt --word_timestamps True --max_line_width 42 --max_line_count 2`

The appropriate-length subtitles will be written to `filename.srt`.

(Source: `whisper --help`)

**Reply by Francoyy** [reply]
Interesting, thanks for the information!
I actually installed whisper with the command `pip install -U openai-whisper` and I don't have this feature. I see that `--max_line_width` was added [2 months ago](https://github.com/openai/whisper/commit/43940fc9780cb91c4f94899755b4648e19d7b977).

I wonder how often or what it would take to have the pip package openai-whisper updated?

**Reply by Francoyy** [reply]
Solved with using 
`pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git`
Thanks for the help, very excited about this feature!

**Reply by Francoyy** [reply]
One note: in the log output, the final sentence truncation doesn't show up. It is only in the final SRT file that the truncation is showing.
It could be nice to also see the truncation on the go, as for long videos it takes up to a few hours to get the final SRT file.

**Reply by ryanheise** [reply]
This shouldn't truncate anything, it should only wrap the tokens. If it's indeed truncating, it sounds like a serious bug and it would help if you could provide an audio file to help reproduce it.

If you actually just mean wrapping, the text you see on screen as it's processing is just the raw output of the segments while Whisper is inferring them. It's not the actual final output. Those segments then undergo further processing later on to produce output in particular formats. By default, actually all of the different formats are written simultaneously, and the writer architecture in theory supports things like lookahead which couldn't be outputted progressively as easily. In the case of the JSON output format, you need a complete JSON tree to be constructed before you can convert that to JSON. In theory, it would be possible to redesign the architecture to support progressive output in some cases, but I'm not sure it's worth it, as I think the current on-screen display of raw progress is useful for what it is, i.e. giving you the progress.

**Reply by Francoyy** [reply]
Understood, thanks for the details! Yeah, I didn't mean truncate, I guess I meant 'split' over multiple timestamps according to the max width number. Love this tool.

**Reply by charl-ie** [reply]
Is this also available as a python command? This would really help me out

**Reply by ryanheise** [reply]
https://github.com/openai/whisper/discussions/98#discussioncomment-6459910

**Comment by rexsateesh** [comment]
I have tested this code and working fine for me.

```python
import whisper
from whisper.utils import get_writer 

audio = './audio.mp3'
model = whisper.load_model(model='small')
result = model.transcribe(audio=audio, language='en', word_timestamps=True, task="transcribe")

# Set VTT Line and words width
word_options = {
    "highlight_words": False,
    "max_line_count": 1,
    "max_line_width": 42
}
vtt_writer = get_writer(output_format='vtt', output_dir='./')
vtt_writer(result, audio, word_options)
```
